Add install script telemetry section (#331)
shuttle-hq/shuttle-docs: _snippets/telemetry.mdx

Reviewer: All of this could be wrapped in a `<Info>` if that looks good, and no need to use a snippet for something that is only used in one place :)

> The install script collects anonymous telemetry data to help improve the product. You can opt out by setting any of these environment variables to a truthy value:

Reviewer: ```suggestion
The install script collects anonymous telemetry data to help improve the product. You can opt out by setting any of these environment variables to `1` or `true`:
```

- `DO_NOT_TRACK`
- `DISABLE_TELEMETRY`
- `SHUTTLE_DISABLE_TELEMETRY`

> 
> Telemetry is also automatically disabled in CI environments when the `CI` environment variable is set to `true`.

Reviewer: ```suggestion
- `CI`
```
